docs/MAINTAINERS.md

```diff
@@ -1,52 +1,180 @@
 Maintaining Etcher
 ==================
 
-This document is meant to serve as a guide for maintainers to perform common
-tasks.
+This document is meant to serve as a guide for maintainers to perform common tasks.
 
-Preparing a new version
------------------------
+Releasing
+---------
 
-- Bump the version number in the `package.json`'s `version` property.
+### Release Types
 
-- Bump the version number in the `npm-shrinkwrap.json`'s `version` property.
+- **snapshot** (default): Describe me!
+- **production**: Describe me too!
 
-- Add a new entry to `CHANGELOG.md` by running `make changelog`.
+### Flight Plan
 
-- Update `screenshot.png` so it displays the latest version in the bottom
-right corner.
+#### Preparation
+
+- [Prepare the new version](#preparing-a-new-version)
+- [Generate build artefacts](#generating-binaries) (binaries, archives, etc.)
+- [Draft a release on GitHub](https://github.com/resin-io/etcher/releases)
+    - Upload build artefacts to GitHub release draft
+
+#### Testing
+
+- Test the prepared release and build artefacts properly on **all supported operating systems** to prevent regressions that went uncaught by the CI tests
```

Comment:
Maybe worth linking to my manual tests doc?

Reply:
Added